internal/dnfjson/cache.go

```diff
@@ -136,15 +136,20 @@ func (r *rpmCache) shrink() error {
 	nDeleted := 0
 	for idx := 0; idx < len(r.repoRecency) && r.size >= r.maxSize; idx++ {
 		repoID := r.repoRecency[idx]
-		repo := r.repoElements[repoID]
+		nDeleted++
+		repo, ok := r.repoElements[repoID]
+		if !ok {
+			// cache inconsistency?
```

Comment:
Should we not panic here? How could this happen?

Reply:
> How could this happen?

You're right.  This should only happen as a result of a bug in the code that uses these things and they're all private, so a panic would catch a programming error.